### ISSUE TITLE
A Go version of FLIF

### ISSUE THREAD

**Issue by DonaldTsang** [FLIF-hub/FLIF#527]
As FLIF's specification is taking its final form, maybe it is time to write a Go/Java/C# implementation of FLIF?
Go is known to be easy to read, and does checking like Java (but is faster than Java).
C# could be another replacement, but that would tie itself too close to Microsoft.
Some has said that Crystal or Nim could be a good replacement, but that would be too niche.

**Comment by SylvainDebaudringhien** [comment]
C# being open source, I would hardly say "tied"...
But of course I'm a C# dev, so definitely biased... ;)

**Comment by DonaldTsang** [comment]
@SylvainDebaudringhien in that case, the more "reference/base" implementations the better.
Also when I say Go implementation I don't mean something like https://github.com/chrisfelesoid/go-flif

**Comment by lehitoskin** [comment]
tl;dr - Be the change you want to see in the world.


An implementation in another language seems outside the scope of this project. C/++ is the reference implementation that others are welcome to read and use to make their own versions, possibly in other languages. Unfortunately, saying "we should do x" or even "it's time to do y" is easy to say, but the main developers obviously have plenty of other things to take care of, so asking them to add another item to the list --- especially one as huge as reimplementing the library in another language they may not know --- isn't very realistic. Unless you, specifically, have the time and will to do this, the odds of this being even started are rather slim. I'm not saying this to be a killjoy, it'd be really cool to have versions of the library entirely in another language, but there's no one who has the chops (or the time) to do it. But what do I know --- I'm just some dude on the Internet.

**Comment by SimantoR** [comment]
On this regard, once the licensing issue is resolved, I’m pretty sure ImageSharp (a managed image manipulation library written in C#) will try to incorporate that into their supported formats. But i agree that it is not the responsibility of the authors to do so.

I would love to help implement that into C#.

Regards
Simanto

**Comment by DonaldTsang** [comment]
@SylvainDebaudringhien @SimantoR thank you you two for making a C# implementation, that would help.

**Comment by AdamFrisby** [comment]
Hi - we're looking into a C# port as well, could I potentially sponsor someone to help make this happen? 

If you think you can do it, we'd be happy to put a bounty on the job. Email me adam@sinewavecompany.com if you're interested :)

**Comment by DonaldTsang** [comment]
https://github.com/chrisfelesoid/go-flif
https://github.com/matthiaskrgr/flifcrush
